src/shogun/io/visitors/FilterVisitor.h

```diff
@@ -0,0 +1,174 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Viktor Gal, Giovanni De Toni
+ */
+
+#ifndef SHOGUN_FILTERVISITOR_H
+#define SHOGUN_FILTERVISITOR_H
```

Comment:
not sure what this is about ....

Reply:
forgot to rebase?

Reply:
I need a way to iterate over all parameters of certain type, so this is to help with that.